polkadot/runtime/parachains/src/assigner_bulk/mod.rs

```diff
@@ -45,15 +46,21 @@ use sp_std::prelude::*;
 
 pub use pallet::*;
 
-pub trait WeightInfo {}
+pub trait WeightInfo {
+	fn assign_core() -> Weight;
+}
 
 /// A weight info that is only suitable for testing.
 pub struct TestWeightInfo;
 
 /// Fraction expressed as a nominator with an assumed denominator of 57,600.
 pub type PartsOf57600 = u16;
 
-impl WeightInfo for TestWeightInfo {}
+impl WeightInfo for TestWeightInfo {
+	fn assign_core() -> Weight {
```

Comment:
This should be parameterized at least on the number of assignments. Are there other variables (parameters or state) that could influence the weight?

Reply:
Done! 

It doesn't seem like there are any other parameter or state influences, unless you count parameters that cause error cases. In error cases the resource costs are lower, so idk that it makes sense to account for them when benchmarking for the worst case.

All state accesses happen once each and are O(1)